iree/samples/static_library/static_library_demo.c

```diff
@@ -0,0 +1,221 @@
+        iree_hal_executable_import_provider_null(), iree_allocator_system(),
+        &library_loader);
+  }
+
+  iree_task_executor_t* executor = NULL;
+  if (iree_status_is_ok(status)) {
+    iree_task_executor_create_from_flags(iree_allocator_system(), &executor);
+  }
+
+  // Create the device and release the executor and loader afterwards.
+  if (iree_status_is_ok(status)) {
+    iree_hal_task_device_create(iree_make_cstring_view("dylib"), &params,
+                                executor, 1, &library_loader,
+                                iree_allocator_system(), device);
+  }
+  iree_task_executor_release(executor);
+  iree_hal_executable_loader_release(library_loader);
+
+  return status;
+}
+
+iree_status_t Run() {
+  iree_status_t status = iree_ok_status();
+
+  // Instance configuration (this should be shared across sessions).
+  iree_runtime_instance_options_t instance_options;
+  iree_runtime_instance_options_initialize(IREE_API_VERSION_LATEST,
+                                           &instance_options);
+  iree_runtime_instance_options_use_all_available_drivers(&instance_options);
+  iree_runtime_instance_t* instance = NULL;
+
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_instance_create(&instance_options,
+                                          iree_allocator_system(), &instance);
+  }
+
+  // Create dylib device with static loader.
+  iree_hal_device_t* device = NULL;
+  if (iree_status_is_ok(status)) {
+    status = create_device_with_static_loader(&device);
+  }
+  iree_vm_module_t* hal_module = NULL;
+  if (iree_status_is_ok(status)) {
+    status =
+        iree_hal_module_create(device, iree_allocator_system(), &hal_module);
+  }
+
+  // Session configuration (one per loaded module to hold module state).
+  iree_runtime_session_options_t session_options;
+  iree_runtime_session_options_initialize(&session_options);
+  iree_runtime_session_t* session = NULL;
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_session_create_with_device(
+        instance, &session_options, device,
+        iree_runtime_instance_host_allocator(instance), &session);
+  }
+
+  // Load bytecode module from the embedded data. Append to the session.
+  const struct iree_file_toc_t* module_file_toc =
+      iree_samples_static_library_simple_mul_create();
+  iree_const_byte_span_t module_data =
+      iree_make_const_byte_span(module_file_toc->data, module_file_toc->size);
+  iree_vm_module_t* bytecode_module = NULL;
+  if (iree_status_is_ok(status)) {
+    status = iree_vm_bytecode_module_create(module_data, iree_allocator_null(),
+                                            iree_allocator_system(),
+                                            &bytecode_module);
+  }
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_session_append_module(session, bytecode_module);
+  }
+
+  // Lookup the entry point function call.
+  const char kMainFunctionName[] = "module.simple_mul";
+  iree_runtime_call_t call;
```

Comment:
this is still uninitialized memory
(again, it may be easier to switch to IREE_CHECK_OK here instead of trying to do the proper memory management status checks - and would make this much more readable as a sample)

Reply:
added a memset to address the issue.